src/httpserver.cpp

```diff
@@ -677,3 +697,9 @@ std::string urlDecode(const std::string &urlEncoded) {
     }
     return res;
 }
+
+void GetRPCServerStats(RPCServerStats* rpc_server_stats) {
+    assert(rpc_server_stats != nullptr);
+    std::lock_guard<std::recursive_mutex> rpcserverstats_guard(g_rpcserverstats_mutex);
+    memcpy(rpc_server_stats, &g_rpcserverstats, sizeof(RPCServerStats));
```

Comment:
An alternative is to return a copy of `g_rpcserverstats`.

Reply:
I think this is probably the right approach here given that it is a fairly lightweight data structure.